File: Week 5 Strings in Java/Program 5.java
/*Java function (using function) that checks whether a given number is prime or not
(Note : function type - with arguments with return values)*/

  
import java.util.Scanner;

public class CheckPrime{
    public static void main(String[] args){
        Scanner sc=new Scanner(System.in);
        System.out.print("Enter a number:"); //input number from user
        int number=sc.nextInt();
        boolean isPrime=checkPrime(number);
        if(isPrime==true) //prints if condition is true
        {
            System.out.println("Number is prime.");
        }
        else //prints if condition is not true
        {
            System.out.println("Number is not prime.");
        }
        sc.close();
    }
    //function to check if number is prime or not
    public static boolean checkPrime(int num)
    {
        if(num<=1){
            return false; 
        }        
        for (int i=2;i*i<=num;i++){
            if(num%i==0){
                return false; 
            }
        }       
        return true; 
    }
}

File: Week 10 Python Program's to implement functions/Program 5.py
#python program to Check if a Number is Even or Odd and also check whether it is Prime or not.

def is_even(num):
    return num % 2 == 0

def is_prime(num):
    if num <= 1:
        return False
    elif num <= 3:
        return True
    elif num % 2 == 0 or num % 3 == 0:
        return False
    i = 5
    while i * i <= num:
        if num % i == 0 or num % (i + 2) == 0:
            return False
        i += 6
    return True

def main():
    try:
        num = int(input("Enter a number:"))
        if is_even(num):
            print(f"{num} is even.")
        else:
            print(f"{num} is odd.")

        if is_prime(num):
            print(f"{num} is prime.")
        else:
            print(f"{num} is not prime.")
    except ValueError:
        print("Invalid input")

main()
